apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsLinkControl.tsx

```diff
@@ -0,0 +1,198 @@
+  JSON_FORMS_RANKING.LinkControl,
+  and(
+    isStringControl,
+    schemaMatches((schema) => schema.format === "link"),
+  ),
+)
+
+const RadioCard = ({ children, ...rest }: PropsWithChildren<UseRadioProps>) => {
+  const { getInputProps, getRadioProps } = useRadio(rest)
+
+  return (
+    <Box
+      as="label"
+      _first={{
+        "> div": {
+          borderLeftRadius: "base",
+        },
+      }}
+      _last={{
+        "> div": {
+          borderRightRadius: "base",
+        },
+      }}
+    >
+      <input {...getInputProps()} />
+      <Box
+        {...getRadioProps()}
+        cursor="pointer"
+        border="1px solid"
+        borderColor="base.divider.strong"
+        bgColor="utility.ui"
+        px="1rem"
+        py="0.5rem"
+        mx={0}
+        _checked={{
+          bgColor: "interaction.muted.main.active",
+          color: "interaction.main.default",
+          borderColor: "interaction.main.default",
+        }}
+        textTransform="none"
+        fontWeight={500}
+        lineHeight="1.25rem"
+      >
+        {children}
+      </Box>
+    </Box>
+  )
+}
+
+interface RadioContentProps {
+  selectedLinkType: string
+  data: string
+  handleChange: (value: string) => void
+}
+
+const RadioContent = ({
+  selectedLinkType,
+  data,
+  handleChange,
+}: RadioContentProps) => {
+  switch (selectedLinkType) {
+    case "page":
+      return (
+        <Input
+          type="text"
+          value={data}
+          onChange={(e) => handleChange(e.target.value)}
+          placeholder="Page permalink"
+        />
+      )
+    case "external":
+      return (
+        <Input
+          type="text"
+          value={data}
+          onChange={(e) => handleChange(e.target.value)}
+          placeholder="https://www.isomer.gov.sg"
+        />
+      )
+    case "file":
+      return (
+        <Input
+          type="text"
+          value={data}
+          onChange={(e) => handleChange(e.target.value)}
+          placeholder="File link"
+        />
+      )
+    case "email":
+      return (
+        <Input
+          type="text"
+          value={data.startsWith("mailto:") ? data.slice("mailto:".length) : ""}
+          onChange={(e) => handleChange(`mailto:${e.target.value}`)}
+          placeholder="test@example.com"
+        />
+      )
+    default:
+      return <></>
+  }
+}
+
+export function JsonFormsLinkControl({
+  data,
+  label,
+  handleChange,
+  path,
+  description,
+  required,
+}: ControlProps) {
+  const [selectedLinkType, setSelectedLinkType] = useState("page")
+
+  const handleLinkTypeChange = (value: string) => {
+    setSelectedLinkType(value)
+    handleChange(path, "")
+  }
+
+  const { getRootProps, getRadioProps } = useRadioGroup({
+    name: "link-type",
+    defaultValue: "page",
+    onChange: handleLinkTypeChange,
+  })
+
+  const dataString = data && typeof data === "string" ? data : ""
+
+  return (
+    <Box py={2}>
```

Comment:
supernit: suggest using `rem` instead of numbers for clarity

Reply:
Updated in 5e33dc3f352940442a3b6f798764b8dd3fef94d1.